### PR TITLE
InstaLite ImportPosition

### PR DESCRIPTION
**Pull request by iri031**
Implemented importPosition. Addresses #268.

- [x] Review
- [x] Deployment [here](https://github.com/Instadapp/dsa-connectors/pull/269#issuecomment-1264695944)
- [x] Add to SDK https://github.com/Instadapp/dsa-connect/pull/234
- [x] Enable
- [x] Update docs

### CODE REVIEW

File: contracts/mainnet/connectors/instaLite_v2/main.sol [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		setUint(setIds[0], stEthAmt_);
+		setUint(setIds[1], wethAmt_);
+
+		eventName_ = "LogImport(address,uint256,uint256,uint256,uint256,uint256[],uint256[])";
+		eventParam_ = abi.encode(flashTkn_,flashAmt_,route_,stEthAmt_,wethAmt_,getIds,setIds);
+	}
+}
+
+contract ConnectV2InstaLiteV2 is InstaLiteConnector {
+	string public constant name = "InstaLite-v2.0";
```
Comment (thrilok209) [inline contracts/mainnet/connectors/instaLite_v2/main.sol]:
```suggestion
	string public constant name = "InstaLite-Import-v1.0";
```

File: contracts/mainnet/connectors/instaLite_v2/main.sol [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+		setUint(setIds[0], stEthAmt_);
+		setUint(setIds[1], wethAmt_);
+
+		eventName_ = "LogImport(address,uint256,uint256,uint256,uint256,uint256[],uint256[])";
+		eventParam_ = abi.encode(flashTkn_,flashAmt_,route_,stEthAmt_,wethAmt_,getIds,setIds);
+	}
+}
+
+contract ConnectV2InstaLiteV2 is InstaLiteConnector {
```
Comment (thrilok209) [inline contracts/mainnet/connectors/instaLite_v2/main.sol]:
```suggestion
contract ConnectV2InstaLiteImport is InstaLiteConnector {
```

File: contracts/mainnet/connectors/instaLite_v2/main.sol [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+			"importPosition(address,uint256,uint256,address,uint256,uint256)",
+			flashTkn_,
+			flashAmt_,
+			route_,
+			address(this),
+			stEthAmt_,
+			wethAmt_
+		);
+
+		Address.functionDelegateCall(
```
Comment (thrilok209) [inline contracts/mainnet/connectors/instaLite_v2/main.sol]: @Richa-iitr why delegate call?

Comment (thrilok209) [reply contracts/mainnet/connectors/instaLite_v2/main.sol]: @Richa-iitr You can directly call the function. No need of delegate call

Comment (iri031) [reply contracts/mainnet/connectors/instaLite_v2/main.sol]: msg.sender will change on direct call. We need msg.sender as the user address, right?

File: contracts/mainnet/connectors/instaLite_v2/main.sol (hunk 1 above)
Comment (iri031) [reply contracts/mainnet/connectors/instaLite_v2/main.sol]: done

File: contracts/mainnet/connectors/instaLite_v2/main.sol (hunk 2 above)
Comment (iri031) [reply contracts/mainnet/connectors/instaLite_v2/main.sol]: done

File: contracts/mainnet/connectors/instaLite_v2/main.sol (hunk 3 above)
Comment (iri031) [reply contracts/mainnet/connectors/instaLite_v2/main.sol]: My bad understood.

Comment (iri031) [reply contracts/mainnet/connectors/instaLite_v2/main.sol]: done

File: contracts/mainnet/connectors/instaLite_v2/main.sol [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+			flashTkn_,
+			flashAmt_,
+			route_,
+			address(this),
+			stEthAmt_,
+			wethAmt_
+		);
+
+		setUint(setIds[0], stEthAmt_);
+		setUint(setIds[1], wethAmt_);
```
Comment (thrilok209) [inline contracts/mainnet/connectors/instaLite_v2/main.sol]: @Richa-iitr We need to store iETH amount instead of stEthAmount and wethAmount

Comment (iri031) [reply contracts/mainnet/connectors/instaLite_v2/main.sol]: updated.

**Review by KaymasJain — DISMISSED** [review]

File: contracts/mainnet/connectors/instaLite_import/main.sol [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+		uint256[] memory getIds,
+		uint256 setId
+	) external returns (string memory eventName_, bytes memory eventParam_) {
+		uint256 stEthAmt_ = getUint(getIds[0], stEthAmt_);
+		uint256 wethAmt_ = getUint(getIds[1], wethAmt_);
+		stEthAmt_ = stEthAmt_ == type(uint256).max
+			? astethToken.balanceOf(msg.sender)
+			: stEthAmt_;
+
+		astethToken.approve(address(iEth), stEthAmt_);
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/instaLite_import/main.sol]: Approval must be given to iETH DSA address as attached in the issue.

`iETH.importPosition` is right. No changes there but here in approval address will change.

Comment (iri031) [reply contracts/mainnet/connectors/instaLite_import/main.sol]: Done.

**Review by KaymasJain — APPROVED** [review]

**Comment by iri031** [comment]
### Deployment [Mainnet]

**Name**: `LITE-B`
**Addr**: [0xee7afECd937a7a133e421719D790563F871410E7](https://etherscan.io/address/0xee7afECd937a7a133e421719D790563F871410E7#code)
**tx** [0x6420ae1be088452cb903ebfbf6881437685a8c6c6a0e037cb999877a3d380db8](https://etherscan.io/tx/0x6420ae1be088452cb903ebfbf6881437685a8c6c6a0e037cb999877a3d380db8#eventlog)

**Review by thrilok209 — APPROVED** [review]

**Pull request merged**